File: analysis/Bittree-Dmap/tree.cxx
// C++ program to create an empty 
// treeor and push values one
// by one.
#include <iostream>
#include <AMReX_Vector.H>
#include <AMReX_DistributionMapping.H>

using namespace std;
  
int main()
{

    // Create an empty treeor
    amrex::Vector<int> tree; 
 
    tree.push_back(0);
    tree.push_back(1);
    tree.push_back(2);
    tree.push_back(3);
    tree.push_back(4);
    tree.push_back(5);
    tree.push_back(6);
    tree.push_back(7);
    tree.push_back(8);
    tree.push_back(9);
    tree.push_back(10);
    tree.push_back(11);
    tree.push_back(12);
    tree.push_back(13);
    tree.push_back(14);
    tree.push_back(15);

    int ltree = tree.size();
    cout << ltree << "\n";

    int nprocs = 3;

    int div = int(ltree)/nprocs;
    int mod = ltree%nprocs;

    cout << div << "," << mod << "\n";

}

File: analysis/Bittree-Dmap/tree.py
import numpy

tree = numpy.arange(16, 376, 1, dtype=int)

nprocs = 11
div = int(len(tree) / nprocs)
mod = len(tree) % nprocs

print(len(tree), nprocs, div, mod)

dmap_proc = numpy.array([-1] * len(tree))
for proc in range(nprocs):
    dmap_proc[proc * div + min(proc, mod) : (proc + 1) * div + min(proc + 1, mod)] = proc

dmap_tree = numpy.array([-1] * len(tree))
for block in range(len(tree)):
    dmap_tree[block] = 0
